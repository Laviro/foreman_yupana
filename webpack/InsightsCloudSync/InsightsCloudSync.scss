.rh-cloud-insights {
  #main {
    padding: 0;

    #breadcrumb .form-group {
      margin-bottom: 0;

      h1 {
        font-weight: bold;
      }
    }

    .btn-toolbar {
      display: inherit;
      width: 100%;
    }

    .pf-c-pagination {
      padding-right: 0;
    }
  }
}
