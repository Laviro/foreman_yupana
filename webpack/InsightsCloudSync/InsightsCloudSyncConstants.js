import { translate as __ } from 'foremanReact/common/I18n';

export const INSIGHTS_CLOUD_SYNC_SUCCESS = 'INSIGHTS_CLOUD_SYNC_SUCCESS';
export const INSIGHTS_CLOUD_SYNC_FAILURE = 'INSIGHTS_CLOUD_SYNC_FAILURE';

export const INSIGHTS_SYNC_PAGE_TITLE = __('Red Hat Insights cloud sync');
