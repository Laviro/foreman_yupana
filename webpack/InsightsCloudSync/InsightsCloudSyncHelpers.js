import { foremanUrl } from '../ForemanRhCloudHelpers';

export const insightsCloudUrl = path => foremanUrl(`/insights_cloud/${path}`);
