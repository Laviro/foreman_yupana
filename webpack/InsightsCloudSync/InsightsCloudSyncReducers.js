import { combineReducers } from 'redux';
import { reducers as settingsReducers } from './Components/InsightsSettings';

export default {
  InsightsCloudSync: combineReducers({
    ...settingsReducers,
  }),
};
