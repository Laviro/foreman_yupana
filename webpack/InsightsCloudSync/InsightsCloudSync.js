import React from 'react';
import { IntlProvider } from 'react-intl';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import { Button } from 'patternfly-react';
import { INSIGHTS_SYNC_PAGE_TITLE } from './InsightsCloudSyncConstants';

const InsightsCloudSync = ({ syncInsights }) => {
  document.title = INSIGHTS_SYNC_PAGE_TITLE;
  return (
    <IntlProvider locale={navigator.language}>
      <div className="insights-cloud-sync">
        <h1>{__('Red Hat Insights Sync')}</h1>
        <Button bsStyle="primary" onClick={syncInsights}>
          {__('Sync now')}
        </Button>
      </div>
    </IntlProvider>
  );
};

InsightsCloudSync.propTypes = {
  syncInsights: PropTypes.func.isRequired,
};

export default InsightsCloudSync;
