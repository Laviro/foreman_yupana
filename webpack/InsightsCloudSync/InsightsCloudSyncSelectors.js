import URI from 'urijs';
import { selectRouterLocation } from 'foremanReact/routes/RouterSelector';
import {
  selectAPIResponse,
  selectAPIStatus,
  selectAPIError,
} from 'foremanReact/redux/API/APISelectors';
import { INSIGHTS_HITS_API_KEY } from './InsightsCloudSyncConstants';

export const selectQuery = state => selectRouterLocation(state).query;

export const selectSearch = state => {
  const { search = '' } = selectQuery(state);
  return URI.decodeQuery(search);
};

export const selectPage = state => {
  const { page = '1' } = selectQuery(state);
  return Number(page);
};

export const selectPerPage = state => {
  const perPage = selectQuery(state).per_page;
  return perPage && Number(perPage);
};

export const selectQueryParams = state => ({
  page: selectPage(state),
  perPage: selectPerPage(state),
  searchQuery: selectSearch(state),
});

export const selectHasToken = state =>
  selectAPIResponse(state, INSIGHTS_HITS_API_KEY).hasToken;

export const selectStatus = state =>
  selectAPIStatus(state, INSIGHTS_HITS_API_KEY);

export const selectError = state =>
  selectAPIError(state, INSIGHTS_HITS_API_KEY);
