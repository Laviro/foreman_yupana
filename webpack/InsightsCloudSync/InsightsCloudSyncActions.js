import API from 'foremanReact/API';
import { insightsCloudUrl } from './InsightsCloudSyncHelpers';
import {
  INSIGHTS_CLOUD_SYNC_SUCCESS,
  INSIGHTS_CLOUD_SYNC_FAILURE,
} from './InsightsCloudSyncConstants';

export const syncInsights = () => async dispatch => {
  try {
    await API.post(insightsCloudUrl('tasks'));
    dispatch({
      type: INSIGHTS_CLOUD_SYNC_SUCCESS,
      payload: {},
    });
  } catch (error) {
    dispatch({
      type: INSIGHTS_CLOUD_SYNC_FAILURE,
      payload: {
        error: error.message,
      },
    });
  }
};
