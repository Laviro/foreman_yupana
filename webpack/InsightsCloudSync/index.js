import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './InsightsCloudSyncActions';
import InsightsCloudSync from './InsightsCloudSync';

// map state to props
const mapStateToProps = state => ({});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = {};

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(InsightsCloudSync);
