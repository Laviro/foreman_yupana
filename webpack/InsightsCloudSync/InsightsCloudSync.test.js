import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { noop } from 'patternfly-react';

import InsightsCloudSync from './InsightsCloudSync';

const fixtures = {
  render: {
    syncInsights: noop,
  },
};

describe('InsightsCloudSync', () =>
  testComponentSnapshotsWithFixtures(InsightsCloudSync, fixtures));
