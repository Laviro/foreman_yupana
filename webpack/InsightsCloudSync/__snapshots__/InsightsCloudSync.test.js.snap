// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsCloudSync render 1`] = `
<IntlProvider
  locale="en-US"
>
  <div
    className="insights-cloud-sync"
  >
    <h1>
      Red Hat Insights Sync
    </h1>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      Sync now
    </Button>
  </div>
</IntlProvider>
`;
